install.packages("seqinr", repos="http://R-Forge.R-project.org")
library(seqinr)
library(stringr)

#####This script manually translates the sequence and makes it into amino acid.
#####It was used as a practice file

data_input_aligned = read.delim("/Users/user/Documents/OneDrive - Nexus365/PhD/Campy_Analysis_ALL/Data/removed_n_43697.txt",header = FALSE)
#I took the header >43697 and \n with bash command
#By making header = FALSE it makes sure that the text file is a value
check <- unlist(data_input_aligned) #This function makes the list to an integer
list_aligned <- as.character(check) #This function makes the integer to a character

aligned <- s2c(list_aligned) #s2c makes the long string into separate individual letters
#cleaned_aligned <- str_replace_all(aligned,"[\r\n]" , "") 
#data_input_aligned <- rm('X.43697')

aligned_amino <- translate(seq = aligned) #translating the nucleotide to amino acid sequence
aligned_data_matrix <- data.matrix(aligned_amino) #This allows us to see the number of sequence in matrix format


#Method to make the aligned file 
data_input_nonaligned <- "ATGGAGAATATTTTTAGCAAAGATTCTGATATTGAACTTGTAGATATAGAAAATTCTATA
AAAAGTAGTTATTTAGACTATTCTATGAGTGTTATTATAGGTCGTGCTTTGCCTGACGCA
AGAGATGGTTTAAAGCCTGTTCATAGAAGAATTTTATATGCTATGCAAAATGATGAGGCA
AAAAGTAGAACAGATTTTGTCAAATCAGCCCGTATAGTGGGTGCTGTTATAGGTCGTTAT
CACCCACATGGAGATACAGCAGTTTATGATGCTTTGGTTAGAATGGCTCAAGATTTTTCT
ATGAGATATCCAAGTATTACAGGACAAGGCAACTTTGGATCTATAGATGGTGATAGTGCC
GCTGCGATGCGTTATACTGAAGCAAAAATGAGTAAACTTTCTCATGAGCTTTTAAAAGAT
ATAGATAAAGATACGGTCGATTTTGTTCCAAATTATGATGGTTCAGAAAGCGAACCTGAT
GTTTTACCTTCTAGGGTTCCAAATTTATTATTAAATGGTTCAAGTGGTATAGCTGTAGGT
ATGGCGACAAACATCCCACCTCATAGTTTAAATGAGTTGATAGATGGACTTTTATATTTG
CTTGATAATAAAGATGCAAGCCTAGAAGAGATTATGCAGTTTATCAAAGGTCCAGATTTT
CCAACAGGTGGAATAATTTATGGTAAAAAAGGTATTATAGAAGCTTATCGCACAGGGCGT
GGTCGCGTGAAAGTGCGAGCTAAAACTCATATTGAAAAAAAGACAAATAAAGATGTTATT
GTTATCGATGAGCTTCCTTATCAAACCAATAAAGCTAGGCTTATAGAGCAGATTGCAGAG
CTTGTTAAAGAAAGGCAAATTGAAGGAATATCTGAAGTAAGAGATGAGAGCAATAAAGAA
GGAATCCGCGTTGTTATAGAGCTTAAACGTGAGGCTATGAGTGAAATTGTTTTAAATAAT
CTATTTAAATCTACCACTATGGAAAGTACTTTTGGTGTGATTATGTTGGCAATTCATAAT
AAAGAACCTAAAATTTTCTCTTTGTTGGAACTTTTAAATCTTTTCTTAACTCATAGAAAA
ACAGTTATTATTAGAAGAACGATTTTTGAACTTCAAAAGGCAAGAGCAAGAGCTCATATT
TTAGAAGGTCTTAAAATTGCACTTGATAATATAGATGAAGTGATTGCTTTAATTAAAAAT
AGTTCTGATAATAATACCGCAAGAGATTCTTTAGTAGCTAAATTTGGTCTTAGTGAGCTT
CAAGCCAATGCTATTTTAGATATGAAACTTGGTCGTTTAACAGGACTTGAAAGAGAAAAA
ATCGAAAATGAACTTGCAGAATTAATGAAAGAAATTGCAAGACTTGAAGAAATTTTAAAA
AGTGAAACCTTGCTTGAAAATTTAATTCGCGATGAATTAAAAGAAATTAGAAGTAAATTT
GATGTGCCACGTATTACTCAAATTGAAGATGATTACGATGATATTGATATTGAAGATTTG
ATTCCTAATGAAAATATGGTTGTAACTATCACACATCGTGGTTATATTAAGCGTGTGCCT
AGTAAACAATATGAAAAACAAAAACGAGGTGGAAAAGGAAAATTAGCCGTTACGACTTAT
GATGATGATTTTATAGAAAGTTTCTTTACGGCAAATACACATGATACGCTTATGTTTGTA
ACAGATCGTGGACAGCTTTATTGGCTTAAAGTTTATAAAATTCCTGAAGGCTCAAGAACG
GCTAAAGGAAAAGCAGTGGTAAATCTTATCAATTTACAAGCTGAAGAAAAAATCATGGCT
ATTATTCCAACCACGGATTTTGATGAGAGCAAATCTTTATGTTTCTTTACTAAAAATGGT
ATTGTAAAGCGTACAAATTTGAGTGAATATCAAAATATCAGAAGTGTAGGAGTTAGAGCG
ATCAACTTGGATGAAAATGATGAGTTGGTAACTGCTATTATTGTTCAAAGAGATGAAGAT
GAAATTTTTGCCACTGGTGGTGAAGAAAATTTAGAAAATCAAGAAATTGAAAATTTAGAT
GATGAAAATCTTGAAAATGAAGAAAGTGTAAGCACACAAGGTAAAATGCTCTTTGCAGTA
ACCAAAAAAGGTATGTGTATCAAATTCCCACTTGCTAAAGTGCGTGAAATCGGCCGTGTA
AGTCGTGGGGTGACGGCTATTAAGTTTAAAGAGAAAAATGACGAATTAGTAGGTGCAGTT
GTTATAGAAAATGATGAGCAAGAAATTTTAAGCATAAGTGCAAAAGGTATAGGAAAACGC
ACCAATGCTGGAGAATATAGATTGCAAAGCAGAGGTGGTAAGGGTGTAATTTGTATGAAA
CTTACAGAAAAAACCAAAGATCTTATTAGCGTAGTTATAGTAGATGAAACTATGGATTTA
ATGGCTCTTACAAGTTCAGGTAAGATGATACGTGTTGATATGCAAAGCATTAGAAAAGCA
GGGCGTAATACGAGTGGTGTCATTGTAGTTAATGTGGAAAATGACGAGGTGGTTAGCATC
GCTAAGTGTCCTAAAGAGGAAAATGACGAGGATGAGTTAAGCGATGAAAACTTTGGTTTA
GATTTGCAATAA
"
non_aligned <- s2c(data_input_nonaligned)
cleaned_non_aligned <- str_replace_all(non_aligned,"[\r\n]","")

non_aligned_amino <- translate(seq = cleaned_non_aligned)
non_aligned_data_matrix <- data.matrix(non_aligned_amino)






e <- "GCTAAGTGTCCTAAAGAGGAAAATGACGAGGATGAGTTAAGCGATGAAAACTTTGGTTTA
GATTTGCAATAA"
str_replace_all(e, "[\r\n]" , "")

